referral.go

```diff
@@ -187,3 +196,78 @@ func (c *Client) createEasypostCreditCard(ctx context.Context, referralCustomerA
 	err = client.post(ctx, "credit_cards", creditCardRequest, &out)
 	return
 }
+
+// BetaAddPaymentMethod adds Stripe payment method to referral customer.
+func (c *Client) BetaAddPaymentMethod(stripeCustomerId string, paymentMethodReference string) (out *PaymentMethodObject, err error) {
+	return c.BetaAddPaymentMethodWithContext(context.Background(), stripeCustomerId, paymentMethodReference)
+}
+
+// BetaAddPaymentMethodWithContext performs the same operation as BetaAddPaymentMethod, but allows
+// specifying a context that can interrupt the request.
+func (c *Client) BetaAddPaymentMethodWithContext(ctx context.Context, stripeCustomerId string, paymentMethodReference string) (out *PaymentMethodObject, err error) {
+	return c.BetaAddPaymentMethodWithPrimaryOrSecondaryAndContext(ctx, stripeCustomerId, paymentMethodReference, PrimaryPaymentMethodPriority)
+}
+
+// BetaAddPaymentMethodWithPrimaryOrSecondary adds Stripe payment method to referral customer with
+// PaymentMethodPriority parameter.
+func (c *Client) BetaAddPaymentMethodWithPrimaryOrSecondary(stripeCustomerId string, paymentMethodReference string, primaryOrSecondary PaymentMethodPriority) (out *PaymentMethodObject, err error) {
+	return c.BetaAddPaymentMethodWithPrimaryOrSecondaryAndContext(context.Background(), stripeCustomerId, paymentMethodReference, primaryOrSecondary)
+}
+
+// BetaAddPaymentMethodWithPrimaryOrSecondaryAndContext performs the same operation as BetaAddPaymentMethodWithPrimaryOrSecondary, but allows
+// specifying a context that can interrupt the request.
+func (c *Client) BetaAddPaymentMethodWithPrimaryOrSecondaryAndContext(ctx context.Context, stripeCustomerId string, paymentMethodReference string, primaryOrSecondary PaymentMethodPriority) (out *PaymentMethodObject, err error) {
+	primaryOrSecondaryEndpoint := ""
+
+	switch primaryOrSecondary {
```

Comment:
Since we're reusing this primary/secondary switch case in multiple places in our app now, we should consider making this a separate helper function.

Reply:
added